src/app/api/auth/[...nextauth]/authOptions.ts

```diff
@@ -31,16 +32,20 @@ export const authOptions: AuthOptions = {
   },
   callbacks: {
     async signIn({ account }) {
-      const guildResp = await fetch(
-        'https://discord.com/api/users/@me/guilds',
-        {
-          headers: {
-            Authorization: `Bearer ${account?.access_token}`,
-          },
+      const response = await fetch('https://discord.com/api/users/@me/guilds', {
+        headers: {
+          Authorization: `Bearer ${account?.access_token}`,
         },
-      );
-      // TODO: Catch possible errors, this can be an array or an error object.
-      const guilds = await guildResp.json();
+      });
+      interface FetchError {
+        message: string;
+        code: number;
+      }
+      const guilds: DiscordGuildObject[] | FetchError = await response.json();
+      if (!(guilds instanceof Array)) {
+        return false;
```

Comment:
Por ahora le puse `return false` en caso de un error de fetch y un redirect en caso de que no sea un miembro de FEC (Discord).
Les parece mejor hacer redirect a `/unauthorized` tambien cuando falle el fetch? 

[Diferencias - Sign in callback](https://next-auth.js.org/configuration/callbacks#sign-in-callback)